src/test/java/emissary/core/channels/AbstractSeekableByteChannelTest.java

```diff
@@ -40,65 +45,76 @@ void testConstrutorValidSize() throws IOException {
-            Assertions.assertThrows(ClosedChannelException.class, () -> sbc.read(ByteBuffer.allocate(5)));
-            Assertions.assertThrows(ClosedChannelException.class, () -> sbc.position());
-            Assertions.assertThrows(ClosedChannelException.class, () -> sbc.position(5));
+            assertFalse(sbc.isOpen());
+            assertThrows(ClosedChannelException.class, () -> sbc.size());
+            assertThrows(ClosedChannelException.class, () -> sbc.read(ByteBuffer.allocate(5)));
+            assertThrows(ClosedChannelException.class, () -> sbc.position());
+            assertThrows(ClosedChannelException.class, () -> sbc.position(5));
         }
     }
 
     @Test
     void testImmutable() throws IOException {
         try (SeekableByteChannel sbc = new TestSeekableByteChannel(10)) {
-            Assertions.assertThrows(NonWritableChannelException.class, () -> sbc.truncate(5));
-            Assertions.assertThrows(NonWritableChannelException.class, () -> sbc.write(ByteBuffer.allocate(5)));
+            assertThrows(NonWritableChannelException.class, () -> sbc.truncate(5));
+            ByteBuffer bb = ByteBuffer.allocate(5);
+            assertThrows(NonWritableChannelException.class, () -> sbc.write(bb));
         }
     }
 
     @Test
     void testPosition() throws IOException {
         try (SeekableByteChannel sbc = new TestSeekableByteChannel(10)) {
-            Assertions.assertEquals(0, sbc.position());
-            Assertions.assertThrows(IllegalArgumentException.class, () -> sbc.position(-1));
-            Assertions.assertEquals(sbc, sbc.position(0));
-            Assertions.assertEquals(0, sbc.position());
-            Assertions.assertEquals(sbc, sbc.position(5));
-            Assertions.assertEquals(5, sbc.position());
-            Assertions.assertEquals(sbc, sbc.position(15));
-            Assertions.assertEquals(15, sbc.position());
+            assertEquals(0, sbc.position());
+            assertThrows(IllegalArgumentException.class, () -> sbc.position(-1));
+            assertEquals(sbc, sbc.position(0));
+            assertEquals(0, sbc.position());
+            assertEquals(sbc, sbc.position(5));
+            assertEquals(5, sbc.position());
+            assertEquals(sbc, sbc.position(15));
+            assertEquals(15, sbc.position());
         }
     }
 
     @Test
     void testRead() throws IOException {
         try (SeekableByteChannel sbc = new TestSeekableByteChannel(10)) {
-            Assertions.assertThrows(NullPointerException.class, () -> sbc.read(null));
+            assertThrows(NullPointerException.class, () -> sbc.read(null));
 
             final ByteBuffer byteBuffer = ByteBuffer.allocate(4);
 
-            Assertions.assertEquals(0, sbc.position());
-            Assertions.assertEquals(4, sbc.read(byteBuffer));
+            assertEquals(0, sbc.position());
+            assertEquals(4, sbc.read(byteBuffer));
```

Comment:
Should also include a test for which there are fewer "content" bytes available in the channel than the size of the byte buffer.  For instance, for this specific channel, set the position to `sbc.size() - 2`, and assert that `sbc.read(byteBuffer)` returns 2.  That helps validate the behavior of the `maxBytesToRead` implementation.

Reply:
Also now addresses in latest commit.